packages/docusaurus-theme-classic/src/theme/CodeBlock/WordWrapButton/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useCallback, useState, useEffect} from 'react';
+import clsx from 'clsx';
+import {translate} from '@docusaurus/Translate';
+import type {Props} from '@theme/CodeBlock/WordWrapButton';
+
+import styles from './styles.module.css';
+
+export default function WordWrapButton({
+  codeBlockRef,
+  className,
+}: Props): JSX.Element | null {
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isCodeScrollable, setIsCodeScrollable] = useState<boolean>(false);
+  const toggleWordWrapCode = useCallback(() => {
+    setIsEnabled((value) => !value);
+
+    const codeElement = codeBlockRef.current!.querySelector('code');
+    codeElement?.classList.toggle(styles.codeWithWordWrap!);
+  }, [codeBlockRef]);
+  const updateCodeIsScrollable = useCallback(() => {
+    const {scrollWidth, clientWidth} = codeBlockRef.current!;
+    setIsCodeScrollable(scrollWidth > clientWidth);
+  }, [codeBlockRef]);
+  const title = translate({
+    id: 'theme.CodeBlock.wordWrapToggle',
+    message: 'Toggle word wrap',
+    description:
+      'The title attribute for toggle word wrapping button of code block lines',
+  });
+
+  useEffect(() => {
+    updateCodeIsScrollable();
+
+    window.addEventListener('resize', updateCodeIsScrollable);
+
+    return () => {
+      window.removeEventListener('resize', updateCodeIsScrollable);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!isCodeScrollable) {
+    return null;
+  }
```

Comment:
all this code is quite technical and shouldn't be there

The button should preferably always render a button, and users swizzling it should be able to change its display/icon without having to manage this technical code. 

I don't think the button should either handle a ref of parent code block. It's the parent that should decide if the toggle button should be displayed, not the button itself. 

I'd suggest moving this complexity to a custom hook in theme-common, and using the hook inside `<CodeBlock>` instead `<WordWrapButton>`